### PR TITLE
 milestone2-add-and-remove

### PR DESCRIPTION
**Pull request by Enoisong**
### In this milestone, I:

-  Implement a function for adding a new task (add a new element to the array).
-  Implement a function for deleting a task (remove an element from the array).
-  Implement a function for editing task descriptions.
-  Implement all changes to the To Do List to be saved in local storage.

### CODE REVIEW

File: src/todos.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  editTodo(todoId, todoDescription) {
+    const newData = this.list.map((todo) => {
+      if (todo.id === todoId) {
+        return { ...todo, description: todoDescription };
+      }
+      return todo;
+    });
+    localStorage.setItem('todos', JSON.stringify(newData));
+  }
+}
```
Comment (ghost) [inline src/todos.js]: - [ ] Good work up to this point but there are some crucial aspects of the application missing
  - When editing and deleting tasks in your app, the changes are not saved in the local storage. Any changes should be saved in the local storage to ensure that consistent data is displayed when the page is refreshed. Kindly address that.

**Review by frank1738 — CHANGES_REQUESTED** [review]
Hi @Enoisong , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

✅ Correct use of Github workflow
✅ No linter error


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

N/A
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/todos.js (hunk 1 above)
Comment (rloterh) [reply src/todos.js]: This issue is well resolved! ✔️

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Enobong,  

Your project is complete! 🏁 There is nothing else to say other than... it's time to merge it 🍾 Congratulations!  🎉 💯 🌟 

## Highlights 🎯 
🎲 CRUD feature correctly implemented!

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

**You can also consider:**
- Let's improve the look of the ReadMe by properly formatting the selected content under the _Future Features_ section, as seen below:

image

Implementing the above change(s) will make your ReadMe/ repository stand out even better! 😉

Cheers..Happy coding!💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**